sdk/core/core/test/cmocka/test_az_credential_client_secret.c

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <az_credentials.h>
+#include <az_credentials_internal.h>
+#include <az_http.h>
+#include <az_http_internal.h>
+#include <az_span.h>
+
+#include <stddef.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+
+#include <cmocka.h>
+
+#include <_az_cfg.h>
+
+static void test_credential_client_secret(void** state)
+{
+  (void)state;
+
+  az_credential_client_secret credential = { 0 };
+  // init credential_credentials struc
+  assert_true(az_succeeded(az_credential_client_secret_init(
+      &credential,
+      AZ_SPAN_FROM_STR("TenantID"),
+      AZ_SPAN_FROM_STR("ClientID"),
+      AZ_SPAN_FROM_STR("ClientSecret"))));
+
+  assert_true(az_succeeded(
+      _az_credential_set_scopes((_az_credential*)&credential, AZ_SPAN_FROM_STR("Scopes"))));
+
+  _az_http_pipeline pipeline = (_az_http_pipeline){
+    ._internal = {
+      .p_policies = {
+        {._internal = { .process = az_http_pipeline_policy_credential, .p_options = &credential, }, },
+        {._internal = { .process = az_http_pipeline_policy_transport, .p_options = NULL, }, },
+      },
+    },
+  };
+
+  az_span const expected_responses[] = {
+    AZ_SPAN_FROM_STR("HTTP/1.1 200 OK\r\n\r\nResponse1"),
+    AZ_SPAN_FROM_STR("HTTP/1.1 200 OK\r\n\r\nResponse2"),
+    AZ_SPAN_FROM_STR("HTTP/1.1 200 OK\r\n\r\nResponse3"),
+    AZ_SPAN_FROM_STR("HTTP/1.1 200 OK\r\n\r\nResponse4"),
+  };
+
+  az_span const request_url = AZ_SPAN_FROM_STR("https://www.microsoft.com/test/request");
+  for (int i = 0; i < 4; ++i)
+  {
+    az_result ignore = { 0 };
+
+    uint8_t header_buf[500] = { 0 };
+    uint8_t body_buf[500] = { 0 };
+    _az_http_request request = { 0 };
+    ignore = az_http_request_init(
+        &request,
+        &az_context_app,
+        az_http_method_get(),
+        request_url,
+        az_span_size(request_url),
+        AZ_SPAN_FROM_BUFFER(header_buf),
+        AZ_SPAN_FROM_BUFFER(body_buf));
+
+    az_http_response response = { 0 };
+    uint8_t response_buf[500] = { 0 };
+    ignore = az_http_response_init(&response, AZ_SPAN_FROM_BUFFER(response_buf));
+
+#ifdef _az_MOCK_ENABLED
+    will_return(__wrap_az_platform_clock_msec, ((i / 2) + 1) * 100000000);
```

Comment:
Add a comment explaining the meaning behind this calculation. why are we doing `i / 2 + 1` and what does the multiplication by 100000000 mean?

Reply:
this is what az_platform_clock_msec() will return. I will comment.

Reply:
Fixed